### ISSUE TITLE
Response reading is randomly blocked on alpine 3.8

### ISSUE THREAD

**Issue by chapurlatn** [aio-libs/aiohttp#3676]
## Long story short

Some specific call response reading are blocking on alpine 3.8 (docker python:3.7-alpine3.8).
This happens only on some URLs (and response) and the response is always read unless the last byte.

```python
async with aiohttp.ClientSession() as session:
        async with session.get(ur) as response:
            print(response.status) # ok
            print(await reponse.text()) # never ends, nothing printed!
```

This is reproduced while streaming chunks using dedicated iterator `response.content.iter_chunks()`.

```python
async with aiohttp.ClientSession() as session:
        async with session.get(ur) as response:
            print(response.status) # ok
            async for data, _ in response.content.iter_chunks(): # never go out the loop
              print(data) # all the body is printed
              #  !!! on debian, an empty chunkis printed at the end and it goes out the loop !!!
```

But it does not happen anymore by iterating "manually" over chunk and checking the end status using `response.content.readchunk()`

```python
async with aiohttp.ClientSession() as session:
        async with session.get(ur) as response:
            print(response.status) # ok
            (chunk, end) = await response.content.readchunk()
            print(chunk)
            while end:
                (chunk, end) = await response.content.readchunk()
                print(chunk)
           # this is fully working !
```

First remark: A curl call on the same alpine image works.

Second remark: I was able to fix it by updating the streams.py module in my image after the install and upgrading the chunk iterator (I will join the PR ASAP).

## Expected behaviour

All layers returns the body quickly as aiohttp does usually.

## Actual behaviour

High level chunk reading blocks sometime.

## Steps to reproduce

```
$ docker run -it python:3.7-alpine3.8 /bin/sh
# pip install aiohttp
# python3 -X dev
> 
```
Copy Paste the following python script:
```
import aiohttp
import asyncio
import logging

logging.basicConfig(level=logging.DEBUG)

async def main():
    url = '...'
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            print(response.status)
            print(await reponse.text()) # blocks sometime

loop = asyncio.get_event_loop()
loop.set_debug(True)
loop.run_until_complete(main())
```

Unfortunately, I was able to reproduce the behaviour only on a request which returns really secured content at my client's office :-(

I can provide generic information like the response body size or any other information you may need to confirm a potential "bug".

Here is a curl call partially censured, hope this help.

```
# curl -sS -v -k -H 'Authorization: Bearer CONFIDENTIALITY_CANT_SHARE' https://server.com/url/which.fail/only/on/alpine/3.8 > /dev/null
*   Trying 10.11.12.13...
* TCP_NODELAY set
* Connected to server.com (10.11.12.13) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
} [245 bytes data]
* TLSv1.2 (IN), TLS handshake, Server hello (2):
{ [58 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [2037 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [300 bytes data]
* TLSv1.2 (IN), TLS handshake, Request CERT (13):
{ [107 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Certificate (11):
} [7 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [37 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-CHACHA20-POLY1305
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CONFIDENTIALITY_CANT_SHARE
*  start date: CONFIDENTIALITY_CANT_SHARE
*  expire date: CONFIDENTIALITY_CANT_SHARE
*  issuer: CONFIDENTIALITY_CANT_SHARE
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x5590493fa5e0)
> GET /url/which.fail/only/on/alpine/3.8 HTTP/2
> Host: server.com
> User-Agent: curl/7.61.1
> Accept: */*
> Authorization: Bearer CONFIDENTIALITY_CANT_SHARE
> 
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200 
< audit-id: CONFIDENTIALITY_CANT_SHARE
< cache-control: no-store
< content-type: application/json
< date: Wed, 03 Apr 2019 16:03:24 GMT
< 
{ [5926 bytes data]
* Connection #0 to host server.com left intact
```

## Your environment

docker python:3.7-alpine3.8 fails
docker python:3.7.2-slim-stretch works (like centos or any other serious distro)

**Comment by socketpair** [comment]
`HTTP/2` ?! aiohttp does not support that. Or I miss something ?

**Comment by kxepal** [comment]
Suddenly, we don't /:

**Comment by chapurlatn** [comment]
Right, the curl call is just to provide some information about the call.
I can provide the curl call using HTTP1.1:

```
curl --http1.1 -sS -v -k -H 'Authorization: Bearer CONFIDENTIALITY_CANT_SHARE' https://server.com/url/which.fail/only/on/alpine/3.8 > /dev/null
*   Trying 10.11.12.13...
* TCP_NODELAY set
* Connected to server.com (10.11.12.13) port 443 (#0)
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
} [242 bytes data]
* TLSv1.2 (IN), TLS handshake, Server hello (2):
{ [64 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [2037 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [300 bytes data]
* TLSv1.2 (IN), TLS handshake, Request CERT (13):
{ [107 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Certificate (11):
} [7 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [37 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-CHACHA20-POLY1305
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CONFIDENTIALITY_CANT_SHARE
*  start date: CONFIDENTIALITY_CANT_SHARE
*  expire date: CONFIDENTIALITY_CANT_SHARE
*  issuer: CONFIDENTIALITY_CANT_SHARE
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
> GET /url/which.fail/only/on/alpine/3.8 HTTP/1.1
> Host: server.com
> User-Agent: curl/7.61.1
> Accept: */*
> Authorization: Bearer CONFIDENTIALITY_CANT_SHARE
> 
< HTTP/1.1 200 OK
< Audit-Id: CONFIDENTIALITY_CANT_SHARE
< Cache-Control: no-store
< Content-Type: application/json
< Date: Wed, 03 Apr 2019 16:27:15 GMT
< Transfer-Encoding: chunked
< 
{ [998 bytes data]
* Connection #0 to host server.com left intact

```

**Comment by socketpair** [comment]
I have made different changes to streams in aiohttp and I want definitely know how to reproduce. Anyway, in order changes to be merged you have to make tests that reproduce the bug.

Also, `end` indicator means that this is the last subchunk of real HTTP chunk. it may be empty.

**Comment by socketpair** [comment]
@chapurlatn can you test without SSL layer ? (i.e. http:// instead of https://) ? and give logs from curl. You may replace sensitive data with `*`

**Comment by chapurlatn** [comment]
@socketpair I will try tomorrow since I can't fetch the failing service from home... But I'm pretty sure I could not since the service is a private cloud pretty well secured.

I'm gonna try to find out what's the case I encountered by digging the code.

**Comment by socketpair** [comment]
@chapurlatn `curl --trace-ascii output.txt https://you-secret.url`. Next, replace (not remove) secret bytes with stars and give this file to me

**Comment by chapurlatn** [comment]
@socketpair 

`curl --http1.1 -s -o /dev/null --trace-ascii ~/tmp/output.txt -H 'Authorization: Bearer ******************************************' https://*********.****.***.****.***/****/*****************/**/**********/******************/*****************/***************`

[output.txt](https://github.com/aio-libs/aiohttp/files/3042556/output.txt)

**Comment by socketpair** [comment]
Well. What I see:
1. This is a chunked transfer with one data chunk of length 6702 bytes
2. I don't see last five bytes of cleartext ( hidden under stars, but all lengths match, so I conclude it exists)
3. Transfer is split to sub-chunks
The last chunk is important in context we speak about. Please clarify. Maybe your webserver send it in a wrong way..I need last 8 bytes from last `<= Recv data`.

So anyway, iterating it using `iter_chunks()` should give:

either
```
(<992 bytes>, False)
(<2374 bytes>, False)
(<535 bytes>, False)
(<2801 bytes>, False)
(<empty>, True)
```
or 
```
(<992 bytes>, False)
(<2374 bytes>, False)
(<535 bytes>, False)
(<2801 bytes>, True)
```

**Comment by chapurlatn** [comment]
@socketpair I'm really sorry but I can't reproduce anymore today.

Yesterday, several people were able to reproduce on a Docker cluster infra, locally using docker, using several system (Ubuntu, Debian, Mac). Each time, request passes using curl, failed using alpine 3.8 + python3 + aiohttp.

Today, I tried on a fresh alpine: can't reproduce, I tried using the original failing images: can't reproduce. I even redeployed the failing version of our service on the production Docker cluster: can't reproduce either.
As if something has been fixed on the remote system. Somthing which failed using aiohttp+python3+alpine3.8 and not using other system.

I'll try again later but at this point, I think we could close the issue.

Again, many thanks for the time you took and for aiohttp which really rocks!
